contentcuration/contentcuration/frontend/channelEdit/views/ImportFromChannels/SearchOrBrowseWindow.vue

```diff
@@ -213,6 +429,133 @@
             throw error;
           });
       }),
+      handleAboutRecommendations() {
+        this.showAboutRecommendations = true;
+      },
+      closeAboutRecommendations() {
+        this.showAboutRecommendations = false;
+      },
+      handleViewMoreRecommendations() {
+        if (!this.recommendationsLoadingError) {
+          const pageSize = this.recommendationsPageSize;
+          const currentIndex = this.recommendationsCurrentIndex;
+          const nextIndex = currentIndex + pageSize;
+          if (this.showViewMoreRecommendations) {
+            this.displayedRecommendations = this.recommendations.slice(0, nextIndex);
+            this.recommendationsCurrentIndex = nextIndex;
+            this.showViewMoreRecommendations = this.moreRecommendationsAvailable;
+          } else if (this.shouldLoadOtherRecommendations) {
+            this.loadRecommendations(true);
+          } else {
+            this.displayedRecommendations = [
+              ...this.recommendations,
+              ...this.otherRecommendations.slice(0, nextIndex - this.recommendations.length),
+            ];
+            this.recommendationsCurrentIndex = nextIndex;
+            if (this.showNoDirectMatches) {
+              this.showNoDirectMatches = this.moreRecommendationsAvailable;
+            }
+            if (this.showShowOtherResources) {
+              this.showShowOtherResources = this.moreRecommendationsAvailable;
+            }
+          }
+        } else {
+          this.loadRecommendations(this.recommendationsBelowThreshold);
+        }
+      },
+      async loadRecommendations(belowThreshold = false) {
+        if (this.isAIFeatureEnabled) {
+          this.recommendationsLoading = true;
+          this.recommendationsLoadingError = false;
+          try {
+            return this.fetchRecommendations(belowThreshold).then(async recommendations => {
+              return await this.fetchRecommendedNodes(recommendations).then(nodes => {
```

Comment:
Unnecessary `.then` chaining on promises in this `async loadRecommendations` function.